compiler-cli/src/build.rs

```diff
@@ -48,9 +54,12 @@ pub fn main(options: Options, manifest: Manifest) -> Result<Built> {
         compiler.compile()?
     };
 
-    match perform_codegen {
-        Codegen::All | Codegen::DepsOnly => cli::print_compiled(start.elapsed()),
-        Codegen::None => cli::print_checked(start.elapsed()),
-    };
+    if !no_print_progress {
+        match perform_codegen {
+            Codegen::All | Codegen::DepsOnly => cli::print_compiled(start.elapsed()),
+            Codegen::None => cli::print_checked(start.elapsed()),
+        };
+    }
```

Comment:
Move these onto the telemetry object please, don't want to have any conditional logic for this.